ui/perfherder/graphs/GraphsViewControls.jsx

```diff
@@ -26,6 +26,13 @@ export default class GraphsViewControls extends React.Component {
     updateStateParams({ highlightedRevisions: newRevisions });
   };
 
+  extractMeasurementUnitsSet = testData => {
+    const measurementUnits = testData.map(
+      testDetails => testDetails.measurementUnit || null,
```

Comment:
This is going to include a null value in the returned array if `measurementUnit` is undefined.

Reply:
I needed to ensure I have consistent placeholder values, for when real measurement units are missing. Indeed, this could have been done better, right after the data is fetched from the endpoints & not in code sections like these ones.
I'm addressing this.

Reply:
I'm not following the placeholder aspect since it's being returned as a Set so there's still the possibility of returning a null value (I think even with the changes you've made). :)

Reply:
You mean use empty string as default instead of `null`? If yes, that is indeed better.

Reply:
I meant more that perhaps using `filter` based on a truthy value might be better, so you don't have a Set being returned with a falsey value (like null).